test/functional/insert.test.js

```diff
@@ -812,6 +812,7 @@ describe('Insert', function() {
       var configuration = this.configuration;
       var client = configuration.newClient(configuration.writeConcernMax(), { poolSize: 1 });
       client.connect(function(err, client) {
+        test.equal(null, err);
```

Comment:
let's not add new `test.equal` lines, I would prefer we remove them when we encounter them. Let's just use `expect(err).to.not.exist` instead. (this applies to a few changes below)